client/src/api.rs

```diff
@@ -175,6 +175,7 @@ pub struct DiskRequest {
     pub read_only: bool,
     pub key: Option<String>,
     pub gen: u64,
+    pub device: String,
```

Comment:
meta question: Is there anything we want to do to coordinate changes to the client api?

Reply:
I'm not aware if that's an answered question or not. I know that versioning has been discussed a few times, and that there's some relevant RFDs too.

At this stage of development, at least a change like this will cause a compile failure instead of a silent runtime failure. In that vein, I verified that Omicron will build with this change (they use a rev versioned dependency anyway :) ).